flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java

```diff
@@ -207,8 +206,110 @@ private void testRemovingTrailingCR(String lineBreaker, String delimiter) {
 				assertEquals(content, result);
 			}
 
+		} catch (Throwable t) {
+			System.err.println("test failed with exception: " + t.getMessage());
+			t.printStackTrace(System.err);
+			fail("Test erroneous");
+		}
+	}
+
+	@Test
+	public void testUTF16Read() {
+		final String first = "First line";
+		final String second = "Second line";
+
+		try {
+			// create input file
+			File tempFile = File.createTempFile("TextInputFormatTest", "tmp");
+			tempFile.deleteOnExit();
+			tempFile.setWritable(true);
+
+			PrintStream ps = new PrintStream(tempFile, "UTF-16");
+			ps.println(first);
+			ps.println(second);
+			ps.close();
+
+			TextInputFormat inputFormat = new TextInputFormat(new Path(tempFile.toURI().toString()));
+			inputFormat.setCharsetName("UTF-32");
+
+			Configuration parameters = new Configuration();
+			inputFormat.configure(parameters);
+
+//			inputFormat.setDelimiter("\r");
```

Comment:
please add tests to check if the `'\'` case is correctly handled for the different encodings.

Reply:
@fhueske I am sorry to reply to you so late. Thank you very much for your inspection and suggestions. I will modify and adjust them one by one according to your suggestions.